malduck/extractor/extract_manager.py

```diff
@@ -184,39 +91,105 @@ def on_extractor_error(
             traceback.format_exc(),
         )
 
-    def push_file(self, filepath: str, base: int = 0) -> Optional[str]:
+    def push_file(self, filepath: str, base: int = 0) -> List[str]:
         """
         Pushes file for extraction. Config extractor entrypoint.
 
         :param filepath: Path to extracted file
         :type filepath: str
         :param base: Memory dump base address
         :type base: int
-        :return: Family name if ripped successfully and provided better configuration than previous files.
-            Returns None otherwise.
+        :return: List of detected families
         """
         log.debug("Started extraction of file %s:%x", filepath, base)
         with ProcessMemory.from_file(filepath, base=base) as p:
             return self.push_procmem(p, rip_binaries=True)
 
-    def push_config(self, family: str, config: Config) -> Optional[str]:
-        config["family"] = family
-        if family not in self.configs:
-            self.configs[family] = config
-            return family
-        else:
-            base_config = self.configs[family]
-            if is_config_better(base_config, config):
-                log.debug("Config looks better")
+    def match_procmem(self, p: ProcessMemory) -> YaraRulesetMatch:
+        """
+        Performs Yara matching on ProcessMemory using modules
+        bound with current ExtractManager.
+        """
+        matches = p.yarap(self.rules, extended=True)
+        log.debug("Matched rules: %s", list(matches.keys()))
```

Comment:
```suggestion
        log.debug("Matched rules: %s", ",".join(list(matches.keys())))
```
To be explicitly correct